math/primes.go

```diff
@@ -0,0 +1,32 @@
+package math
+
+import (
+	"crypto/rand"
+	"io"
+	"math/big"
+)
+
+// IsSafePrime reports whether p is (probably) a safe prime.
+// The prime p=2*q+1 is safe prime if both p and q are primes.
+// The n parameter is the number of checks for primality test.
+func IsSafePrime(p *big.Int, n int) bool {
```

Comment:
Why make `n` configurable? Perhaps best to just use 20, as [crypto/rand.Prime](https://pkg.go.dev/crypto/rand#Prime) does.

Reply:
removed the parameter.